cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -595,20 +622,81 @@ inline std::array<int64_t, 3> GetIsoCalendar(int64_t arg, Localizer&& localizer)
           static_cast<int64_t>(weekday(ymd).iso_encoding())};
 }
 
+template <typename Duration, typename InType>
+struct ISOCalendarWrapper {
+  static Status Get(const Scalar& in, std::array<int64_t, 3>& iso_calendar) {
```

Comment:
For later: either pass a const reference or a non-const pointer. Non-const references require more care from the caller.